### DISCUSSION TITLE
Need help skipping song when Miniget error occurs

### DISCUSSION THREAD

**Discussion by nicocespede** [Androz2091/discord-player#1352] [Q&A]
Hello everyone, for some time whenever I got an error I've been just destroying the queue and making the bot to stop playing music, but now I'd like it to just skip the song which got the `MinigetError` and go on with the rest of the queued songs.

This is how I refactored my 'error' listener:

```
.on('error', (queue, error) => {
        console.log(error);
        const success = queue.skip();
        queue.metadata.send({
                content: '<@${ids.users.stormer}>',
                embeds: [musicEmbed.setDescription(success ? '⏭️ Hubo un error al reproducir ${"${SONG_NAME}"}. Canción salteada.' : '🛑 Ocurrió un error.').setThumbnail(success ? 'attachment://icons8-end-64.png' : 'attachment://icons8-no-entry-64.png')],
                files: [success ? './assets/thumbs/music/icons8-end-64.png' : './assets/thumbs/music/icons8-no-entry-64.png']
        });
        if (!success)
                queue.destroy();
});
```

However, after this code executes, I just get the `[PlayerError] Cannot play a resource that has already ended.` error.

Is there any way I can do this and skip the song properly?

**Comment by twlite** [comment]
fd208755cf12fee9cd2e9b5bd8683bd04c9289e9 should possibly fix this issue. I will release this update soon in `dev` channel.

**Reply by twlite** [reply]
@nicocespede are you able to handle this in `discord-player@dev`? There is also a new `<Queue>.forceNext()` method to init next track.

**Reply by nicocespede** [reply]
I'll try it and I tell you then!

**Reply by nicocespede** [reply]
Hey @archaeopteryx1, look, I've tried the dev version you told me about, but before that I had to update all my code to work with discord.js v14, cause I was still using v13. The thing is now when it tries to play that specific song that caused the `MinigetError`, it never even starts, or gets to the error, it just stays buffering and does absolutely nothing. I know that it keeps buffering forever because I saw that with the `'debug'` event listener. So what would you recommend I should do?

**Reply by twlite** [reply]
@nicocespede this needs some more investigation

**Reply by Saddam171** [reply]
si
